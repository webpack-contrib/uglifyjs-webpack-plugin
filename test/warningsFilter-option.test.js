import UglifyJsPlugin from '../src/index';
import {
  cleanErrorStack,
  createCompiler,
  compile,
} from './helpers';

describe('when applied with `warningsFilter` option', () => {
  let compiler;

  beforeEach(() => {
    compiler = createCompiler({
      entry: {
        one: `${__dirname}/fixtures/unreachable-code.js`,
        two: `${__dirname}/fixtures/unreachable-code-2.js`,
      },
    });
  });

  it('matches snapshot for a `function` value and `sourceMap` is `true`', () => {
    new UglifyJsPlugin({
      warningsFilter(source) {
        if (/unreachable-code-2\.js/.test(source)) {
          return true;
        }

        return false;
      },
      uglifyOptions: {
        warnings: true,
      },
      sourceMap: true,
    }).apply(compiler);


    return compile(compiler).then((stats) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot('errors');
      expect(warnings).toMatchSnapshot('warnings');

      for (const file in stats.compilation.assets) {
        if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
          expect(stats.compilation.assets[file].source()).toMatchSnapshot(file);
        }
      }
    });
  });

  it('matches snapshot for a `function` value and `sourceMap` is `false`', () => {
    new UglifyJsPlugin({
      warningsFilter(source) {
        if (/unreachable-code-2\.js/.test(source)) {
          return true;
        }

        return false;
      },
      uglifyOptions: {
        warnings: true,
      },
      sourceMap: false,
    }).apply(compiler);


    return compile(compiler).then((stats) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot('errors');
      expect(warnings).toMatchSnapshot('warnings');

      for (const file in stats.compilation.assets) {
        if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
          expect(stats.compilation.assets[file].source()).toMatchSnapshot(file);
        }
      }
    });
  });
});
