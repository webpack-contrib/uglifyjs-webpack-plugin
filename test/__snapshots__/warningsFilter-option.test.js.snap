// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with \`warningsFilter\` option matches snapshot for a \`function\` value and \`sourceMap\` is \`false\`: errors 1`] = `Array []`;

exports[`when applied with \`warningsFilter\` option matches snapshot for a \`function\` value and \`sourceMap\` is \`false\`: manifest.7064adc867fd70846e42.js 1`] = `"!function(r){var n=window.webpackJsonp;window.webpackJsonp=function(e,u,c){for(var f,i,p,a=0,l=[];a<e.length;a++)i=e[a],o[i]&&l.push(o[i][0]),o[i]=0;for(f in u)Object.prototype.hasOwnProperty.call(u,f)&&(r[f]=u[f]);for(n&&n(e,u,c);l.length;)l.shift()();if(c)for(a=0;a<c.length;a++)p=t(t.s=c[a]);return p};var e={},o={2:0};function t(n){if(e[n])return e[n].exports;var o=e[n]={i:n,l:!1,exports:{}};return r[n].call(o.exports,o,o.exports,t),o.l=!0,o.exports}t.m=r,t.c=e,t.d=function(r,n,e){t.o(r,n)||Object.defineProperty(r,n,{configurable:!1,enumerable:!0,get:e})},t.n=function(r){var n=r&&r.__esModule?function(){return r.default}:function(){return r};return t.d(n,\\"a\\",n),n},t.o=function(r,n){return Object.prototype.hasOwnProperty.call(r,n)},t.p=\\"\\",t.oe=function(r){throw console.error(r),r}}([]);"`;

exports[`when applied with \`warningsFilter\` option matches snapshot for a \`function\` value and \`sourceMap\` is \`false\`: one.d321aed3a325448435a0.js 1`] = `"webpackJsonp([0],[function(n,c){}],[0]);"`;

exports[`when applied with \`warningsFilter\` option matches snapshot for a \`function\` value and \`sourceMap\` is \`false\`: two.da1b0e3de955e7df6b6e.js 1`] = `"webpackJsonp([1],[,function(n,c){}],[1]);"`;

exports[`when applied with \`warningsFilter\` option matches snapshot for a \`function\` value and \`sourceMap\` is \`false\`: warnings 1`] = `
Array [
  "UglifyJs Plugin: Dropping unreachable code [one.d321aed3a325448435a0.js:5,41]",
  "UglifyJs Plugin: Dropping unused function foo [one.d321aed3a325448435a0.js:5,9]",
  "UglifyJs Plugin: Dropping unreachable code [two.da1b0e3de955e7df6b6e.js:6,41]",
  "UglifyJs Plugin: Dropping unused function foo [two.da1b0e3de955e7df6b6e.js:6,9]",
]
`;

exports[`when applied with \`warningsFilter\` option matches snapshot for a \`function\` value and \`sourceMap\` is \`true\`: errors 1`] = `Array []`;

exports[`when applied with \`warningsFilter\` option matches snapshot for a \`function\` value and \`sourceMap\` is \`true\`: manifest.7064adc867fd70846e42.js 1`] = `"!function(r){var n=window.webpackJsonp;window.webpackJsonp=function(e,u,c){for(var f,i,p,a=0,l=[];a<e.length;a++)i=e[a],o[i]&&l.push(o[i][0]),o[i]=0;for(f in u)Object.prototype.hasOwnProperty.call(u,f)&&(r[f]=u[f]);for(n&&n(e,u,c);l.length;)l.shift()();if(c)for(a=0;a<c.length;a++)p=t(t.s=c[a]);return p};var e={},o={2:0};function t(n){if(e[n])return e[n].exports;var o=e[n]={i:n,l:!1,exports:{}};return r[n].call(o.exports,o,o.exports,t),o.l=!0,o.exports}t.m=r,t.c=e,t.d=function(r,n,e){t.o(r,n)||Object.defineProperty(r,n,{configurable:!1,enumerable:!0,get:e})},t.n=function(r){var n=r&&r.__esModule?function(){return r.default}:function(){return r};return t.d(n,\\"a\\",n),n},t.o=function(r,n){return Object.prototype.hasOwnProperty.call(r,n)},t.p=\\"\\",t.oe=function(r){throw console.error(r),r}}([]);"`;

exports[`when applied with \`warningsFilter\` option matches snapshot for a \`function\` value and \`sourceMap\` is \`true\`: one.d321aed3a325448435a0.js 1`] = `"webpackJsonp([0],[function(n,c){}],[0]);"`;

exports[`when applied with \`warningsFilter\` option matches snapshot for a \`function\` value and \`sourceMap\` is \`true\`: two.da1b0e3de955e7df6b6e.js 1`] = `"webpackJsonp([1],[,function(n,c){}],[1]);"`;

exports[`when applied with \`warningsFilter\` option matches snapshot for a \`function\` value and \`sourceMap\` is \`true\`: warnings 1`] = `
Array [
  "UglifyJs Plugin: Dropping unreachable code [./test/fixtures/unreachable-code-2.js:1,40] in two.da1b0e3de955e7df6b6e.js",
  "UglifyJs Plugin: Dropping unused function foo [./test/fixtures/unreachable-code-2.js:1,0] in two.da1b0e3de955e7df6b6e.js",
]
`;
